report/report_test.go

```diff
@@ -280,20 +282,37 @@ func TestGenerateReport(t *testing.T) {
 				},
 			},
 		},
+		{
+			description: "Should return error when a dependency source cannot be determined",
+			pkg: []models.Dependency{
+				{
+					Source:   "",
+					Revision: "12345",
+					Name:     "gopkg.in/fake",
+				},
+			},
+			wantReport: models.Report{},
+			wantError: fmt.Errorf("failed to create report object from dependency: { 12345 gopkg.in/fake}: unable to determine repo source for gopkg.in/fake, must add to map"),
+		},
 	}
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
 			gotReport, err := g.BuildReport(productName, test.pkg)
-			if err != nil {
+			if err != nil && test.wantError == nil {
 				t.Errorf("BuildReport failed with errors: %v", err)
 			}
 
+			if err != nil && test.wantError != nil {
+				assert.EqualError(t, err, test.wantError.Error())
+			}
+
```

Comment:
Do we not also need the case of `err == nil && test.wantError != nil`? Aka when the want error is present but none was thrown. I believe that should throw a `t.Errorf`